frontend/src/concepts/pipelines/context/useManageElyraSecret.ts

```diff
@@ -41,9 +45,14 @@ const useManageElyraSecret = (
         return;
       }
       try {
-        const secretValue = JSON.parse(atob(elyraSecret.data?.[ELYRA_SECRET_DATA_KEY] || '{}'));
-        if (secretValue.metadata[ELYRA_SECRET_DATA_TYPE] === 'USER_CREDENTIALS') {
-          // Secret is invalid -- update it
+        const secretValue = JSON.parse(
+          atob(elyraSecret.data?.[ELYRA_SECRET_DATA_KEY] || '{ metadata: {} }'),
+        );
+        const usingOldDataType =
+          secretValue.metadata[ELYRA_SECRET_DATA_TYPE] === 'USER_CREDENTIALS';
+        const usingOldUrl = secretValue.metadata[ELYRA_SECRET_DATA_ENDPOINT].endsWith('/view/');
```

Comment:
> Code looks good...  have you retested this to make sure it works?

I have tested the 2 scenario , 
1. in a new cluster setup, it works as we want.
2. In a old cluster setup, the value doesn't get updated.
i feel the reason is because are not checking the right condition:


```suggestion
        const usingOldUrl = !secretValue.metadata[ELYRA_SECRET_DATA_ENDPOINT].endsWith('/view/');
```

Reply:
ooooh, good catch. That was supposed to be a not. @mlassak please make that change.